nntrainer/layers/preprocess_flip_layer.h

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2020 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * @file	preprocess_flip_layer.h
+ * @date	20 January 2020
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug		No known bugs except for NYI items
+ * @brief	This is Preprocess Random Flip Layer Class for Neural Network
+ *
+ */
+
+#ifndef __PREPROCESS_FLIP_LAYER_H__
+#define __PREPROCESS_FLIP_LAYER_H__
+#ifdef __cplusplus
+
+#include <random>
+
+#include <layer_internal.h>
+#include <tensor.h>
+
+namespace nntrainer {
+
+/**
+ * @class   Preprocess FLip Layer
+ * @brief   Preprocess FLip Layer
+ */
+class PreprocessFlipLayer : public Layer {
+public:
+  /**
+   * @brief     Constructor of Preprocess FLip Layer
+   */
+  template <typename... Args>
+  PreprocessFlipLayer(Args... args) :
+    Layer(args...),
```

Comment:
How about.. nntrainer::augmetation::FilpLayer?

Reply:
Then how about nntrainer::dataset::FlipLayer?
As input normalization, resizing etc will also be added, and they are although for dataset but not augmentation.

Reply:
We could make sub-namespace for others too. I guess, let's start from here.

Reply:
> Then how about nntrainer::dataset::FlipLayer?
> As input normalization, resizing etc will also be added, and they are although for dataset but not augmentation.
currently we support input normalization as an property. I think then it should be the same with this. So.. That could be a layer?

Reply:
With property, the user has to set two configurations - layertype and property. But with property, it can be done with just 1 configuration.
Additionally, it's being coherent with other popular frameworks, but we can make it as a property like activation layer.

Reply:
If we are supporting both ways, then I think it's ok.